public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,309 @@
+    if (this.readyState === XMLHttpRequest.DONE) {
+      if (this.status === 200) {
+        var form = document.createElement('form');
+        form.method = 'POST';
+        if (window.location.hostname === 'payment.edx.org') {
+          form.action = 'https://secureacceptance.cybersource.com/silent/pay';
+        } else {
+          form.action = 'https://testsecureacceptance.cybersource.com/silent/pay';
+        }
+        var ecommerceFormFields = JSON.parse(this.response).form_fields;
+        Object.keys(ecommerceFormFields).forEach(function (key) {
+          var element = document.createElement('input');
+          element.type = 'hidden';
+          element.value = ecommerceFormFields[key];
+          element.name = key;
+          form.appendChild(element);
+        });
+        var cardNumber = document.getElementById('cardnumber').value;
+        cardNumber.type = 'hidden';
+        var cardNumberElement = document.createElement('input');
+        cardNumberElement.value = (cardNumber.match(/\d+/g) || []).join('');
+        cardNumberElement.name = 'card_number';
+        form.appendChild(cardNumberElement);
+
+        // Emma: make sure to test all the possible number options
+        var cardTypeElement = document.createElement('input');
+        cardTypeElement.type = 'hidden';
+        if (cardNumber.charAt(0) === '4') {
+          cardTypeElement.value = '001'; // visa
+        } else if (cardNumber.substring(0, 2) === '34' || cardNumber.substring(0, 2) === '37') {
+          cardTypeElement.value = '003'; // American Express
+        } else if (cardNumber.substring(0, 4) === '6011' || cardNumber.substring(0, 2) === '65' || /^64[4-9]/.test(cardNumber)) {
+          cardTypeElement.value = '004'; // Discover
+        } else if (cardNumber.substring(0, 4) === '2720' || /^5[1-5]/.test(cardNumber) || /^222[1-9]/.test(cardNumber) || /^22[3-9]/.test(cardNumber) || /^2[3-6]/.test(cardNumber) || /^27[0-1]/.test(cardNumber)) {
+          cardTypeElement.value = '002'; // Mastercard
+        } else {
+          redirectToMFE();
+        }
+        cardTypeElement.name = 'card_type';
+        form.appendChild(cardTypeElement);
+
+        var cardCVNElement = document.createElement('input');
+        cardCVNElement.type = 'hidden';
+        cardCVNElement.value = document.getElementById('cardcode').value;
+        cardCVNElement.name = 'card_cvn';
+        form.appendChild(cardCVNElement);
+
+        var cardExpirationElement = document.createElement('input');
+        cardExpirationElement.type = 'hidden';
+        cardExpirationElement.value = document.getElementById('cardmonth').value.concat('-', document.getElementById('cardyear').value);
+        cardExpirationElement.name = 'card_expiry_date';
+        form.appendChild(cardExpirationElement);
+
+        document.body.appendChild(form);
+        form.submit();
+      } else {
+        redirectToMFE();
+      }
+    }
+  };
+  var urlEncodedDataPairs = [];
```

Comment:
Pretty sure this is significantly shorter but equivalent:

```
function encoder(name) {
  returnencodeURIComponent(document.getElementById(name).value)
}
var urlEncodedDataPairs = [
  'basket=' + encodeURIComponent(getBasketId()),
  'first_name=' + encoder('firstname'),
  'last_name=' + encoder('lastname'),
  'address_line1=' + encoder('address1'),
  'address_line2=' + encoder('address2'),
  'city=' + encoder('city'),
  'country=' + encoder('country'),
  'state=' + encoder('state'),
  'postal_code=' + encoder('zip'),
];
```

URL encoding the keys isn't doing anything - the underscore is a valid character in a URL, and the rest is alphanumeric.  I haven't tested this code.